File: src/main/java/me/captainpotatoaim/myplugin/custom_items/railgun/Railgun.java
package me.captainpotatoaim.myplugin.custom_items.railgun;

import me.captainpotatoaim.myplugin.custom_items.CustomItem;
import org.bukkit.Material;
import org.bukkit.enchantments.Enchantment;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;

import static org.bukkit.ChatColor.*;

public class Railgun {
    public static ItemStack getItem() {
        ItemStack railGun = new ItemStack(Material.TRIDENT, 1);
        ItemMeta railGunMeta = railGun.getItemMeta();
        assert railGunMeta != null;
        railGunMeta.setDisplayName(GRAY + "R" + YELLOW + "A" + GRAY + "I" + YELLOW + "L" + GRAY + "G" + YELLOW + "U" + GRAY + "N");
        railGunMeta.addEnchant(Enchantment.INFINITY, 1, true);
        railGun.setItemMeta(railGunMeta);
        CustomItem.setType(railGun, Railgun.class);

        return railGun;
    }
}

File: src/main/java/me/captainpotatoaim/myplugin/custom_items/railgun/Railgun.kt
package me.captainpotatoaim.myplugin.custom_items.railgun;

import me.captainpotatoaim.myplugin.custom_items.CustomItem
import me.captainpotatoaim.myplugin.util.AdventureHelper.createAlternatingColoredText
import net.kyori.adventure.text.format.NamedTextColor.GRAY
import net.kyori.adventure.text.format.NamedTextColor.YELLOW
import org.bukkit.Material
import org.bukkit.enchantments.Enchantment
import org.bukkit.inventory.ItemStack


class Railgun : CustomItem() {
    override fun getItem(count: Int): ItemStack {
        val railGun = ItemStack(Material.TRIDENT, 1)
        val railGunMeta = checkNotNull(railGun.itemMeta)
        railGunMeta.displayName(createAlternatingColoredText("RAILGUN", GRAY, YELLOW))
        railGunMeta.addEnchant(Enchantment.INFINITY, 1, true)
        railGun.setItemMeta(railGunMeta)
        setType(railGun, Railgun::class.java)

        return railGun
    }
}
